package au.gpb;


// ////////////////////////////////////////////////////////////////////////////
//
// Message between ConsoleClient and ConsoleService
//

message Message
{
    enum Type
    {
        message    = 1;
        warning    = 2;
        error      = 3;
    }

	required string txt	= 1;	// Text to be displayed
	required Type type	= 2;	// Type of message
}

message AutoCompletionAlternative
{
    required string label=1;     // Label to show on screen
    required string command=2;   // Text to be compared with the current last word    
    required bool add_space_if_unique=3;  // Add space if this is the unique alternative
}

message ConsolePacket
{
	optional string command     = 1;    // Command introduced by the user
	repeated Message message    = 2;    // Messages to be shown on screen

	optional string auto_complettion_command = 3; // Ask for auto-completion
	repeated AutoCompletionAlternative auto_completion_alternatives = 4; // Answer for auto-completion

	optional string prompt_request = 5;
	optional string prompt = 6;
	
}

// Message used as first packet in connections to a log server

message LogConnectionHello
{
    enum LogConnectionType
    {
        LogProvider    = 1;
        LogProbe       = 2;
    }

	// type of connection
	required LogConnectionType type = 1;

	// optional filters
	optional string filter = 2;


}